### PR TITLE
implemented revealing object names

### PR DESCRIPTION
**Pull request by contriteobserver**
by prefixing each generated child with the name of the source material that caused the object to be created.

addresses #2301

### CODE REVIEW

File: rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java [hunk 1]
```diff
@@ -416,7 +416,12 @@ private void mergeGroupsAsObjects(Object3D object) {
 	}
 
 	private static String generateObjectName() {
-		return "Object" + (int) (Math.random() * 10000);
+		return generateObjectName(null);
+	}
+
+	private static String generateObjectName(String prefix) {
+		if(prefix==null) prefix = "Object";
+		return prefix + (int) (Math.random() * 9e3 + 1e3);
```
Comment (hannesa2) [inline rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: Why not using a timestamp (or UUID) ?

Comment (contriteobserver) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: Excellent idea, I went with a timestamp as the ramifications of that system call is better understood.

File: rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java [hunk 2]
```diff
@@ -421,7 +422,7 @@ private static String generateObjectName() {
 
 	private static String generateObjectName(String prefix) {
 		if(prefix==null) prefix = "Object";
-		return prefix + (int) (Math.random() * 9e3 + 1e3);
+		return prefix + Long.toString(SystemClock.elapsedRealtime());
```
Comment (hannesa2) [inline rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: Keep in mind, this is in nanos, I would go with millis `System.currentTimeMillis()` which is more common to use as unixtimestamp. But I'm fine with it.

Comment (contriteobserver) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: `System.currentTimeMillis()` would not necessarily provide the necessary resolution, as models sometimes duplicate names, and we can parse multiple child objects in a mS

object graph
    -->GROUP default
    	-->Arm.Left2597618548
    	-->Arm.Right2597618769
    	-->Leg.Left2597619008
    	-->Leg.Right2597619185
    	-->Head2597619359
    	-->Standing_Pose

Comment (jonathanchh1) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: I see you guys have now implemented this feature. Is there away to pull these changes or can it be released? I'm working on something really urgent and i would love to have this feature. Thank you!

Comment (hannesa2) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: Really urgent ? Nothing to do, here you go 
`implementation "com.github.contriteobserver:Rajawali:LoadedMtlObjNaming-SNAPSHOT"`
or
`implementation "com.github.contriteobserver:Rajawali:45cf99dc15"`

With jitpack.io you can have a library of any commit. Please see [details](https://jitpack.io/#contriteobserver/Rajawali/45cf99dc15)

Comment (jonathanchh1) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: Yes, definitely urgent! thank you again!

Comment (ToxicBakery) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: If the object names need guaranteed uniqueness, why not a uuid?

File: rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java (hunk 1 above)
Comment (ToxicBakery) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: Seeing this now after my other question. I really think uuid implies the intent vs a random number or nanos usage.

File: rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java (hunk 2 above)
Comment (contriteobserver) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: UUIDs alone are insufficient, as there is a need for name to be revealing. Also, legacy identifiers had the classifier "Object" and the instance label "[0-9][0-9][0-9][0-9]" select-able by character type (letters vs. numbers) it seems appropriate to preserve that ability. 

Also, the overhead of a UUID call is concerning, as it must at least interface with the MAC address, the clock, pseudo-random generator, and a hash system. Was worried a UUID call might slow down file parsing.

Open to ideas if anyone has a suggestion.

Comment (ToxicBakery) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: Maybe we should consider using an atomic number generator then. Maybe useful as the object IDs should be consistent as long as the model or loading order does not change.

Ie. AtomicInteger getAndIncrement()
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicInteger.html#getAndIncrement--

Comment (contriteobserver) [reply rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java]: Implemented an `AtomicInteger` based identification mechanism, the `PosingDoll` object graph now looks like:

 I/MaterialNamingRenderer.InitScreen: object graph
    -->GROUP default
    	-->Arm.Left1002
    	-->Arm.Right1003
    	-->Leg.Left1004
    	-->Leg.Right1005
    	-->Head1006
    	-->Standing_Pose

**Review by ToxicBakery — APPROVED** [review]

**Pull request merged**